src/generators.rs

```diff
@@ -110,17 +109,17 @@ impl Iterator for GeneratorsChain {
     type Item = StarkPoint;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let mut uniform_bytes = [0u8; STARK_UNIFORM_BYTES];
-        self.reader.read(&mut uniform_bytes);
-
-        // TODO: The inputs to the hash-to-curve algorithm here are public, so we may be able to
-        // optimize this by providing a hash-to-curve method that fails to hide the input
+        let mut low_u256 = [0u8; STARK_POINT_BYTES];
+        keccak256(&self.state, &mut low_u256);
+        self.state.copy_from_slice(&low_u256);
+
+        // Samples a uniformly random point on the curve in a manner that is consistent with the
+        // Cairo implementation. Namely, convert the uniform bytes to a scalar field element, and
+        // multiply by the generator point.
+        // Examples of other such hash-to-curve schemes that do not hide the scalar multiple:
         // https://eprint.iacr.org/2009/226.pdf and https://link.springer.com/chapter/10.1007/978-3-642-14623-7_13
-        // provide examples of such schemes
-        Some(
-            StarkPoint::from_uniform_bytes(uniform_bytes)
-                .expect("error deserializing point from random bytes"),
-        )
+        let scalar = Scalar::from_uniform_bytes(low_u256);
```

Comment:
In your other PR, I suggest moving the changes to this repo. I think this is even more true now that I see how this fits together. It's an odd coupling for the `mpc-stark` library to be extending the digest to 512 bits for no reason relevant to the execution of an MPC. Those changes make more sense here and are more readable in this context